### PR TITLE
Iamtesch/condense arm nodes

### PR DESCRIPTION
**Pull request by iamtesch**
Condense the `arm_node` and `arm_node_action` node 

There was a lot of duplicated functionality here -- these both exposed a different ROS interface to the same robot arm.  Condensed the logic so the different messages are all handled by the same class.

### CODE REVIEW

File: nodes/arm_node.cpp [hunk 1]
```diff
@@ -1,110 +1,247 @@
...
+namespace ros {
+
+class ArmNode {
+public:
+  ArmNode(arm::Arm& arm) : arm_(arm) { }
+
+  // Replan a smooth joint trajectory from the current location through a
+  // series of cartesian waypoints.
+  // xyz positions should be a 3xn vector of target positions
+  void updateCartesianWaypoints(const Eigen::Matrix3Xd& xyz_positions,
```
Comment (rLinks234) [inline nodes/arm_node.cpp]: Awesome. Looks like this is a new overload of `updateCartesianWaypoints`, right? The biggest issue I have with the C++ API is that there are no public facing functions which allow you to specify a fixed size dimension in a matrix (_e.g._, here `Eigen::Matrix3Xd == Eigen::Matrix<double, 3, Dynamic>`)

File: nodes/arm_node.cpp [hunk 2]
```diff
@@ -1,110 +1,247 @@
...
+    // Plan to each subsequent point from the last position
+    // (We use the last position command for smoother motion)
+    Eigen::VectorXd last_position = arm_.getLastFeedback().getPositionCommand();
+
+    // Get joint angles to move to each waypoint
+    for (size_t i = 0; i < num_waypoints; ++i) {
+
+      // Find the joint angles for the next waypoint, starting from the last
+      // waypoint, and save into the position vector.
+      if (end_tip_directions.cols() > 0) {
```
Comment (rLinks234) [inline nodes/arm_node.cpp]: Preferably, this conditional branch should be hoisted outside of a loop, considering the if () branch is evaluated above

File: nodes/arm_node.cpp [hunk 3]
```diff
@@ -1,110 +1,247 @@
...
+    Eigen::Matrix3Xd xyz_waypoints(3, 1);
+    xyz_waypoints(0, 0) = target_xyz_.x() + data.x;
+    xyz_waypoints(1, 0) = target_xyz_.y() + data.y;
+    xyz_waypoints(2, 0) = target_xyz_.z() + data.z;
 
-          // Save the waypoints
-          positions.col(i) = last_position; 
-        }
+    // Replan
+    Eigen::Matrix3Xd tip_directions(3, 0);
```
Comment (rLinks234) [inline nodes/arm_node.cpp]: This seems like a code smell. Why do you have to pass in an empty matrix?

File: nodes/arm_node.cpp [hunk 4]
```diff
@@ -188,23 +326,56 @@ int main(int argc, char ** argv) {
...
     return -1;
   }
 
+  // Load the appropriate gains file
+  {
+    hebi::GroupCommand gains_cmd(arm -> size());
+    if (!gains_cmd.readGains(ros::package::getPath(gains_package) + std::string("/") + gains_file))
+      ROS_ERROR("Could not load arm gains file!");
+    else
+    {
```
Comment (rLinks234) [inline nodes/arm_node.cpp]: formatting's not consistent with above

Comment (iamtesch) [reply nodes/arm_node.cpp]: fixed

File: nodes/arm_node.cpp (hunk 3 above)
Comment (iamtesch) [reply nodes/arm_node.cpp]: fixed

File: nodes/arm_node.cpp (hunk 2 above)
Comment (iamtesch) [reply nodes/arm_node.cpp]: fixed

File: nodes/arm_node.cpp (hunk 1 above)
Comment (iamtesch) [reply nodes/arm_node.cpp]: :+1: yeah -- I hadn't realized there were these overloads.  We should totally use these on the gyros/accels if we're not yet.

Comment (iamtesch) [reply nodes/arm_node.cpp]: Basically, this is a helper method that all the other message/action callbacks reference -- I commented this more clearly and made it private

**Pull request merged**